kolibri_explore_plugin/assets/src/modules/topicsRoot/handlers.js

```diff
@@ -69,6 +69,8 @@ function _fetchCarouselNodes(store) {
         const base = `/topics/${node.channel_id}`;
         if (node.kind === 'topic') {
           node.nodeUrl = `${base}/t/${node.id}`;
+        } else if (node.parent === node.channel_id) {
```

Comment:
Maybe we've been doing this wrong? Instead of the discovery page deciding which page to route, we could pass the ID directly and the template should decide. There could be an option override to avoid the detail page for channels like this. "showInParent" or something.

Reply:
That would be preferable, yeah. I'll get rid of this for now - looking at it with fresh eyes, the real problems here are the Wikipedia channel needs descriptions for its Zim content nodes (which I have now added), and Content.vue always includes a "Next in…" card grid even if there's nothing to see - which makes the page seem rather sparse and broken. Those issues are solvable, albeit out of scope for this branch, and this change kind of muddles it.